test/test_cache_related_cases.py

```diff
@@ -223,3 +223,40 @@ def test_delete_one_parent(self):
         with g.session_scope() as s:
             sample = g.nodes(md.Sample).one()
             assert sample._related_cases == [case2]
+
+    def test_preserve_timestamps(self):
```

Comment:
Can we have one tests that does this same thing but maybe uses ThreadPoolExecutor to test concurrency during updates?

Reply:
> Can we have one tests that does this same thing but maybe uses ThreadPoolExecutor to test concurrency during updates?

I can add that test, though I'm not sure we need it if we're confirming that the shared case is not updated. Do you think we should block the merge on that, or are we good to go without it?

Reply:
Nah, this is fine